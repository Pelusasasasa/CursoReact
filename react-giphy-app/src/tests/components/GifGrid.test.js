import { shallow } from "enzyme"
import React from 'react'
import {GifGrid} from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"
jest.mock('../../hooks/useFetchGifs')

describe("Pruebas en <GifGrid/>",()=>{

    const category = "Boca"


    test('hacer match con el snapchot ', () => {

        useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>)

        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe de mostrat items cunado se carga imagenes useFetchGifs', () => {

        const gifs = [{
            id:"ABC",
            url:"https://localhost/cualquier/cosa.jpg",
            title:"Cualquier Cosa"
        }]
        useFetchGifs.mockReturnValue({
            data:gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category}/>)

        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
    
    
})