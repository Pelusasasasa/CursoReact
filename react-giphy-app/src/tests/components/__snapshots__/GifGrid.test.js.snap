// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> hacer match con el snapchot  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Boca
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
